src/member.rs

```diff
@@ -673,7 +697,27 @@ where
         let (consensus_exit, exit_stream) = oneshot::channel();
         let config = self.config.clone();
         let sh = self.spawn_handle.clone();
+        let (messages_for_network, mut messages_from_recovery) =
+            mpsc::unbounded::<(UnitMessage<H, D, MK::Signature>, Recipient)>();
+        let (responses_newest_tx, responses_newest_rx) = mpsc::unbounded();
+        self.tx_recovery = Some(responses_newest_tx);
         info!(target: "AlephBFT-member", "{:?} Spawning party for a session.", self.index());
+        let n_members = self.n_members;
+        let our_index = self.keybox.index();
+        let recovered_unit = recovery::recover_newest_unit::<H, D, MK::Signature>(
+            n_members,
+            responses_newest_rx,
+            messages_for_network,
+            our_index,
+        )
+        .await;
```

Comment:
Is there a good reason that this recovery blocks all the consensus components and not just the Creator? Or is the latter problematic to implement?

Reply:
How bad would be to have everything as it used to be, but Creator would additionally receive a one_shot channel endpoint as an argument through which it receives the `start_round` :P?